tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/object/ObjectInitializerTest.java

```diff
@@ -191,4 +191,78 @@ public void testInitActionInsideObjectDescriptor() {
         Assert.assertEquals(returns[0].getType().getTag(), TypeTags.STRING);
         Assert.assertEquals(returns[0].stringValue(), "Ballerina");
     }
+
+    @Test(description = "Test invoking '__init' function in a function inside object descriptor.")
+    public void testInitInvocationInsideObject() {
+
+        BValue[] returns = BRunUtil.invoke(compileResult, "testInitInvocationInsideObject");
+
+        Assert.assertEquals(((BInteger) returns[0]).intValue(), 75);
+        Assert.assertEquals(((BInteger) returns[1]).intValue(), 80);
+    }
+
+    @Test(description = "Test invoking '__init' function with args in a function inside object descriptor with args.")
+    public void testInitInvocationInsideObjectWithArgs() {
+
+        BValue[] returns = BRunUtil.invoke(compileResult, "testInitInvocationInsideObjectWithArgs");
```

Comment:
This test is not testing related to args.  This should validate the Grade field instead of marks. (see bal file)

Reply:
Done